Fix target data standardization and facet_nrow warning message

LucieContamin opened the pull request:
- [x] Resolve #21 

- [x] Resolve #19

LucieContamin commented:
I merge the style PR #24 in this branch

annakrystalli commented:
Docs Look nice eyh? 😉

LucieContamin commented:
> Docs Look nice eyh? 😉

It looks very nice! thanks again

harryhoch commented:
So, I hate to be a stickler, but I'm not a fan of this..

"Before plotting, the data might require some preparation (filtering, etc.). These step is skipped in this example, please consult the “Plot Model Projections Output” vignette contained in this package for complete examples."...

Is there any reason not to include the code on this page?

nickreich commented:
Noting that this might be a reason to fast-track adding standard data objects to the package. see #27

nickreich commented:
I am going through my github notifications and see that my review has been requested here. Is this ready for review now, or still WIP?

nickreich commented:
I guess one general comment I have is that if this is a breaking change (which it is, given that argument names are being renamed entirely without backwards compatibility), then I think we should maybe version increment a bit more aggressively? Maybe this could be 1.0?

LucieContamin commented:
It is ready for Review. I am still thinking about the example question, but we might want to solve that question in another PR.

LucieContamin commented:
> I guess one general comment I have is that if this is a breaking change (which it is, given that argument names are being renamed entirely without backwards compatibility), then I think we should maybe version increment a bit more aggressively? Maybe this could be 1.0?

Thanks, that's a good point. It is a breaking change and so a "major" release. I was hesitant to release a first `1.0` version. I will update it

nickreich reviewed (changes requested):
I reviewed all the changes and think the replacement of truth--> target seems thorough and complete. Thanks for doing this!

My two comments that need addressing are 
1. as mentioned in another comment, incrementing the version to 1.0 since it is a "breaking change". Unless others think it's ok to add a breaking change since it's a 0.* release? @annakrystalli any thoughts on this?
2. fixing the lintr check so it passes.

LucieContamin commented:
Thank you for the review.

To answer your comments:
1. Will changing to `0.0.0.9100` be better to indicate the package is still in dev, but the last implementation include a major change?
2. The `lintr` check is about the complexity of the function as I understand. To solve it will require a major redesign of the code. I temporary solution will be to include an exception in the `.lintr` file of the package to temporary ignore that issue and maybe look into it in another PR?

annakrystalli commented:
Re: Versioning, this is a topic I think we should discuss at a meeting for all our packages. (Sorry, it was a footer topic to discuss on my list of hubverse org changes but we decided to postpone and I didn't get to the bottom of the list!)

0.0.0.9*** indicates the package is still under development and version 0.0.1 would be the first stable release of a package after which semantic versioning applies more strictly.

I believe a number of our packages are at that stage, e.g. `hubValidations` and I'm planning to release `hubUtils`, hubAdmin and possibly `hubData` with 0.0.1 version seeing as the functionality is mature enough by now.

So I leave it up to @LucieContamin really to decide whether the functionality is stable enough for 0.0.1 or she still feels a 0.0.0.9* is still more appropriate. Incrementing by 10 instead of 1 might be more appropriate if so.

I suggest we also think about this with respect to `hubEnsembles`.

Hope this helps! Perhaps @bsweger has some more useful insights from industry too.

annakrystalli commented:
Sorry @LucieContamin ! You basically said exactly the same but much more concisely 😆.

LucieContamin commented:
The PR is becoming quite complex so please find here a small summary of the content:
- Fixes #25
- Fixes #19
- to fix `lintr` issues:
  -  update code style
  - simplify function `plot_step_ahead_output()` and internal function `plot_output()` by creating multiple small functions
      - `palette_utils`: contains the internal code for the plot palette
      - `scatter_plot_utils`: contains the internal functions to create a plot (preparation of the data, add trace on a plot, etc.)
      - `subplot_utils`: contains the internal functions to create the output plot (create simple or complex subplot, etc.)
      - `validate_format`: contains the internal functions to validate the input parameters of `plot_step_ahead_output()`
- update the target data related parameters:
  - fixes #21
  - change the default parameters `x_target_col_name = "date"`
  - expect the target data input to have a column `"observation"` (previously `"value"`)
- use the data from hubExamples in the documentation
  - remove `inst` files (previously used for examples)
- update the tests and documentation accordingly

I will prefer to wait before moving to version `0.1`. I would like to solve 2 main issues before:
- finish solving #16 (currently fix only work on static plot)
- #13

annakrystalli reviewed (approved):
Overall I really like the changes in this PR. The modularisation of the functionality makes it a lot easier to follow what's going on!

I've made some quite minor suggestions but overall happy to merge. My one overall suggestion is that, given plots are the main output of the package, it's a good idea to add some visual regression tests with `vdiffr` package: https://github.com/r-lib/vdiffr

Happy to merge this though and leave that as an issue for future effort

LucieContamin commented on R/validate_format.R:
```diff
@@ -0,0 +1,227 @@
+#'
+#' @noRd
+output_type_validation <- function(model_output_data, exp_value) {
+  if (!class(exp_value) %in% class(model_output_data$output_type_id)) {
```
Thanks for the comment, I updated the documentation. 

So currently, as the function is only accepting quantile et median output type, it's expected to have numeric in the `output_type_id` column. `exp_value` is the numeric vector with the the `output_type_id` value that are going to be used for the plot: for example: `0.5`, '0.975`, 0.025`, etc. The `exp_value` is created internally in the `plot_step_ahead_model_output()` function.  I force the column to be in the `numeric` because it causes issue if it's not. 

Does that answer your question?

annakrystalli replied:
Sort of. I guess my question is why compare classes of the two objects and not just check whether output type Id is numeric and if not coerce it?

LucieContamin replied:
To be honest, I was thinking of when we will have other format (for example with sample) but as we are not there yet. I will simplify it. Thank you again for noticing it!

LucieContamin commented:
@annakrystalli thank you very much for your review!

So, I hope I fix the `output_type_validation()` issue. I also accepted your suggestion, thank you for that!

In another [PR](https://github.com/Infectious-Disease-Modeling-Hubs/example-complex-scenario-hub/pull/12), we notice that the function was having an issue with empty columns in the `model_output_data` parameter, so I just added a small check and warning output in the `plot_prep_data()` function (in the scatter_plot_utils.R script). I updated the News accordingly.

Please let me know if any other issues or questions! Thanks.

annakrystalli commented:
Great! I'm happy for you to merge. 🎉🎉🎉🚀
